apps/antalmanac/src/components/Map/Map.tsx

```diff
@@ -146,6 +141,14 @@ export default function CourseMap() {
           }
         : undefined;
 
+    const hasWeekendCourse = AppStore.getCourseEventsInCalendar().some(
+        (event) => event.start.getDay() === 0 || event.start.getDay() === 6
+    );
+
+    const days = hasWeekendCourse
+        ? ['All', 'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
+        : ['All', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri'];
```

Comment:
Move both of these arrays into global scope so they don't get recomputed on every render. 

_Also change "All" back to ""._

This is used in subsequent lines of code for comparison; all event days will contain an empty string, which is why it's used to represent "All". Any non-empty string will filter out certain days; filtering by "All" would literally filter out all the days since no day starts with "All". You can leave a note if this wasn't clear.

```suggestion
        ? ['', 'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
        : ['', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri'];
```

```ts
const workWeek = ['', 'Mon', 'Tue', ...]
const fullWeek = ['', 'Sun', 'Mon', "Tue', ...]
```

Reply:
```typescript
const workWeek = ['', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri'];
const fullWeek = ['', 'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
...
const days = hasWeekendCourse ? fullWeek : workWeek;
```

I'm not sure if doing that ternary recomputes, thereby making this whole thing pointless, but if it does, then I'd advocate for just filtering out Sat and Sun like I had originally

Reply:
No, this is the best approach. It only does the ternary, but not the array construction. Filtering is worse than this simple recomputation. Being poorly optimized is the hallmark of React.

Reply:
Aye Aye cap'n 🫡